Mathlib/GroupTheory/QuotientGroup/Defs.lean

```diff
@@ -150,17 +162,73 @@ theorem mk_zpow (a : G) (n : ℤ) : ((a ^ n : G) : Q) = (a : Q) ^ n :=
 
 @[to_additive (attr := simp)] lemma map_mk'_self : N.map (mk' N) = ⊥ := by aesop
 
+/--
+The subgroup defined by the class of `1` for a congruence relation on a group.
+-/
+@[to_additive
+"The `AddSubgroup` defined by the class of `0` for an additive congruence relation
+on an `AddGroup`."]
+protected def _root_.Con.subgroup (c : Con G) : Subgroup G where
+  carrier := { x | c x 1 }
+  one_mem' := c.refl 1
+  mul_mem' hx hy := by simpa using c.mul hx hy
+  inv_mem' h := by simpa using c.inv h
+
+@[to_additive (attr := simp)]
+theorem _root_.Con.mem_subgroup_iff {c : Con G} {x : G} :
+    x ∈ c.subgroup ↔ c x 1 := Iff.rfl
+
+@[to_additive]
+instance (c : Con G) : c.subgroup.Normal :=
+  ⟨fun x hx g ↦ by simpa using (c.mul (c.mul (c.refl g) hx) (c.refl g⁻¹))⟩
+
+@[to_additive]
+theorem _root_.Con.subgroup_quotientGroupCon (H : Subgroup G) [H.Normal] :
+    (QuotientGroup.con H).subgroup = H := by
+  ext
+  simp [QuotientGroup.con, leftRel_apply]
+
+@[to_additive (attr := simp)]
+theorem con_subgroup (c : Con G) :
+    QuotientGroup.con c.subgroup = c := by
+  ext x y
+  rw [QuotientGroup.con, Con.rel_mk, leftRel_apply, Con.mem_subgroup_iff]
+  exact ⟨fun h ↦ by simpa using c.mul (c.refl x) (c.symm h),
+    fun h ↦ by simpa using c.mul (c.refl x⁻¹) (c.symm h)⟩
+
+/--
+The normal subgroups correspond to the congruence relations on a group.
+-/
+@[to_additive
+"The normal subgroups correspond to the additive congruence relations on an `AddGroup`."]
+def _root_.Subgroup.orderIsoCon :
+    { N : Subgroup G // N.Normal } ≃o Con G where
+  toFun := fun ⟨N, _⟩ ↦ QuotientGroup.con N
```

Comment:
This will play better with `simps`:
```suggestion
  toFun N := QuotientGroup.con N
```
or perhaps with N.val on the RHS if necessary.

Reply:
It doesn't work either way. Lean complains that it cannot synthesize the fact that `N` is normal.